Upgrade build processes to use PostCSS 8

desrosj opened the pull request:
Trac ticket: https://core.trac.wordpress.org/ticket/57554

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

desrosj committed:
Remove `set-output` from workflow code.

desrosj committed:
Additional adjustments.

desrosj committed:
Temporarily test Slack action.

desrosj committed:
Test Slack notifications.

desrosj committed:
Test right revision of workflow.

desrosj committed:
Correct syntax error.

desrosj committed:
Wrap string in single quotes.

desrosj committed:
Use branch for testing.

desrosj committed:
Remove single quote.

desrosj committed:
Update Core build processes to PostCSS 8.

desrosj committed:
Upgrade Twenty Nineteen to PostCSS 8.

desrosj committed:
Use `autoprefixer` variable.

desrosj committed:
Merge branch 'trunk' of github.com:WordPress/wordpress-develop into upgrade/postcss8

desrosj committed:
Remove unintentional changes.

peterwilsoncc commented on src/wp-content/themes/twentynineteen/style-rtl.css:
```diff
@@ -3006,7 +3006,7 @@ body.page .main-navigation {
 }
 
-.main-navigation .main-menu .menu-item-has-children:not(.off-canvas)[focus-within] > .sub-menu {
+.main-navigation.js-focus-within .main-menu .menu-item-has-children:not(.off-canvas)[focus-within] > .sub-menu, .js-focus-within .main-navigation .main-menu .menu-item-has-children:not(.off-canvas)[focus-within] > .sub-menu {
```
The specificity here is changing from `0,0,6,0` to `0,0,7,0` and the selector is changed from one to two.

Has the default config of focus-within changed as the `js-focus-within` class is new?

desrosj replied:
So I tracked this down. This appears to be something introduced within the [6.0.0 update ](https://github.com/csstools/postcss-plugins/blob/main/plugins/postcss-focus-within/CHANGELOG.md#600-july-8-2022)of `postcss-focus-within`. It looks like version[ 6.1.0 added the ability to prevent this from being added](https://github.com/csstools/postcss-plugins/blob/main/plugins/postcss-focus-within/CHANGELOG.md#610-july-30-2022).

I've gone and added that to the PostCSS configs. But now there's a different issue where the same selectors are added twice.

desrosj replied:
I'll admit, I'm also not really sure the original purpose of including this dependency in the theme to begin with. So it's hard to evaluate whether the new version of the generated CSS or the old version is preferred.

desrosj replied:
Turns out the duplicate classes were a bug in PostCSS and has been resolved in the latest version of `postcss-focus-within` (see csstools/postcss-plugins#835 and csstools/postcss-plugins#840).

After updating, these issues have all been resolved.

peterwilsoncc commented on package.json:
```diff
@@ -24,16 +24,17 @@
 		"last 2 Opera versions"
 	],
 	"devDependencies": {
+		"@lodder/grunt-postcss": "^3.1.1",
```
Does this negate the need for `grunt-postcss` or is that different?

desrosj replied:
Ah! Yes, good find. `grunt-postcss` is [archived and considered abandoned](https://github.com/nDmitry/grunt-postcss). The [PostCSS release notes includes this as the suggested replacement](https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users#grunt). I've removed `grunt-postcss` and updated the lock file.

peterwilsoncc reviewed (commented):
Added a couple of questions inline.

Sorry if they're a bit naive, build tools confuse the heck out of me.

desrosj committed:
Remove `grunt-postcss`.

desrosj committed:
Merge branch 'trunk' into upgrade/postcss8

desrosj committed:
Update related packages to the latest versions.

desrosj committed:
Merge branch 'trunk' into upgrade/postcss8

desrosj committed:
Merge remote-tracking branch 'upstream/trunk' into upgrade/postcss8

desrosj committed:
Merge upstream/trunk.

desrosj committed:
Add `disablePolyfillReadyClass` to PostCSS options.

This returns the generated CSS to the previous state.

desrosj committed:
Add the `disablePolyfillReadyClass` option to the RTL configuration.

desrosj committed:
Update PostCSS dependencies.

desrosj committed:
Merge remote-tracking branch 'upstream/trunk' into upgrade/postcss8

# Conflicts:
#	src/wp-content/themes/twentynineteen/package-lock.json
#	src/wp-content/themes/twentynineteen/package.json

desrosj committed:
Update `postcss-focus-within` to the latest version.

This eliminates duplicate selectors that were being added in the `.7.0.0` update

desrosj committed:
Add RTL updates.

desrosj commented:
Merged in https://core.trac.wordpress.org/changeset/55458.
